pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -187,6 +275,320 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
 	return r.ensureOutputObject(instance, tpcm, reqLogger)
 }
 
+// generateWarningMessage generates a warning message for the user
+// based on the list of deprecated variables and rules that are detected
+// as well as the list of migrated rules that are detected that are not
+// migrated yet
+func generateWarningMessage(variableNeedtoBeDeprecatedList []string, ruleNeedToBeDeprecatedList []string, ruleNeedToBeMigratedList []string) string {
+	var warningMessage string
+	if len(variableNeedtoBeDeprecatedList) > 0 {
+		warningMessage = fmt.Sprintf("The following variables are deprecated and need to be removed from the TailoredProfile: %s", strings.Join(variableNeedtoBeDeprecatedList, ","))
+	}
+	if len(ruleNeedToBeDeprecatedList) > 0 {
+		if warningMessage != "" {
+			warningMessage = fmt.Sprintf("%s\nThe following rules are deprecated and need to be removed from the TailoredProfile: %s", warningMessage, strings.Join(ruleNeedToBeDeprecatedList, ","))
```

Comment:
We could put the new line character at the end of each string, otherwise it's conditional based on the order of operations here, right?